### PR TITLE
Implement redux

### PR DESCRIPTION
**Pull request by tylerconstance**
- Implemented Redux and pulled councilor information from our API endpoint using Axios.
- Brought in a utility to create slugs from our councilors' names.
- Introduced error swallowing in Gulp using `gulp-plumber`

### CODE REVIEW

File: jsx/components/Home.jsx [hunk 1]
```diff
@@ -1,85 +1,52 @@
...
-    "slug":"david-brenerman",
-    "role":"District 5",
-    "citypage":"http://www.portlandmaine.gov"
-  },{
-    "name":"Pious Ali",
-    "slug":"pious-ali",
-    "role":"At Large",
-    "citypage":"http://www.portlandmaine.gov"
+  componentWillMount() {
+    this.props.dispatch(fetchCouncilors())
```
Comment (sethwklein) [inline jsx/components/Home.jsx]: We shouldn't fetch councilors unless we don't have them in the store or we have reason to think they've changed.

File: jsx/main.jsx [hunk 2]
```diff
@@ -2,18 +2,68 @@ import React from 'react';
...
+
+const initialState = {
+  fetching: false,
+  fetched: false,
+  councilors: [],
+  error: null,
+};
+
+const councilorReducer = (state=initialState, action) => {
+  switch(action.type) {
```
Comment (sethwklein) [inline jsx/main.jsx]: `switch (action`... (space before paren)

File: jsx/main.jsx [hunk 3]
```diff
@@ -2,18 +2,68 @@ import React from 'react';
...
+
+const reducers = combineReducers({
+  user: userReducer,
+  councilor: councilorReducer
+})
+
+const store = createStore(reducers, middleware);
+
+store.subscribe(() => {
+  console.log("Store changed", store.getState())
```
Comment (sethwklein) [inline jsx/main.jsx]: Missed debug print?

File: jsx/components/Home.jsx (hunk 1 above)
Comment (tylerconstance) [reply jsx/components/Home.jsx]: I moved the initial data fetching to `main.jsx`—let me know if this looks better!

**Comment by sethwklein** [comment]
👍

**Pull request merged**